pandas/tests/io/excel/test_xlsxwriter.py

```diff
@@ -63,3 +67,29 @@ def test_write_append_mode_raises(ext):
     with ensure_clean(ext) as f:
         with pytest.raises(ValueError, match=msg):
             ExcelWriter(f, engine='xlsxwriter', mode='a')
+
+
+def test_merged_cell_custom_objects(ext):
+    # Test that custom object types residing within merged (grouped)
+    # cells are converted to python data types before being passed to
+    # the xlsxwriter package. Test for issue #27006
+
+    # create a custom object type, and place it in a grouped dataframe
+    pixy = PeriodIndex(['2018', '2018', '2018', '2018',
+                        '2019', '2019', '2019', '2019'], freq='Y')
+    pixq = PeriodIndex(['2018Q1', '2018Q2', '2018Q3', '2018Q4',
+                        '2019Q1', '2019Q2', '2019Q3', '2019Q4'], freq='Q')
+    pixarr = [pixy, pixq]
+    mipix = MultiIndex.from_arrays(pixarr, names=['year', 'quarter'])
+    df = DataFrame(np.random.rand(2, len(mipix)), columns=mipix)
+
+    # write the dataframe to excel
+    try:
```

Comment:
We don't want a try...except within a test here. You will want to write the excel file, read it back in and make assertions about the content using `tm.assert_frame_equal`. If you look at the other tests in test_writers.py should be clearer

Reply:
I fixed this as well. Thank you for the explanation and suggestion here. I looked at some other examples in test_writers.py for guidance, and think I got it. The only little strange bit was that `tm.assert_frame_equal` was initially failing due to the `expected` DataFrame's column MultiIndex being a `PeriodIndex`, and the read-from-file (`result`) column MultiIndex being a standard `Index` containing strings (as that's what `pd.read_excel` returns when it reads it in). I solved for this by just converting the `expected` PeriodIndexes to standard Indexes containing strings. That allows it to past the test - but not 100% if this was kosher so please let me know.